components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -222,7 +222,14 @@ private boolean validateUserCredentials(OAuth2AccessTokenReqDTO tokenReq, Servic
                         (tokenReq.getResourceOwnerUsername()))) {
                     throw new IdentityOAuth2Exception("Authentication failed for " + tenantAwareUserName);
                 }
-                throw new IdentityOAuth2Exception("Authentication failed for " + tokenReq.getResourceOwnerUsername());
+                String username;
+                if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                    // For tenant qualified urls, no need to send fully qualified username in response.
+                    username = tenantAwareUserName;
+                } else {
+                    username = tokenReq.getResourceOwnerUsername();
+                }
```

Comment:
```suggestion
                String username = tenantAwareUserName;
                if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
                    // <suitable comment>
                    username = tokenReq.getResourceOwnerUsername();;
                }
```

Reply:
WDYT? Thinking whether dropping the else block might improve readability...

Reply:
+1

Reply:
done